templates/common/_base/units/machine-config-daemon-pull.service.yaml

```diff
@@ -0,0 +1,28 @@
+name: "machine-config-daemon-pull.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Machine Config Daemon Pull
+  # Make sure it runs only on OSTree booted system
+  ConditionPathExists=/run/ostree-booted
+  BindsTo=ignition-firstboot-complete.service
+  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+  # We only want to run on 4.3 clusters and above; this came from
+  # https://github.com/coreos/coreos-assembler/pull/768
+  ConditionPathExists=/sysroot/.coreos-aleph-version.json
+  # Skip this unit if MCD binary already exists at libexec
+  ConditionPathExists=!/usr/libexec/bin/machine-config-daemon
+  After=ignition-firstboot-complete.service
+  Before=machine-config-daemon-host.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  # Extract machine-config-daemon binary from MCD container
+  ExecStart=/usr/bin/sh -c "while ! /usr/bin/podman pull --authfile=/var/lib/kubelet/config.json '{{ .Images.setupEtcdEnvKey }}'; do sleep 5; done"
+  ExecStart=/usr/bin/sh -c "/usr/bin/podman run --authfile=/var/lib/kubelet/config.json --quiet --net=host --entrypoint=cat '{{ .Images.setupEtcdEnvKey }}' /usr/bin/machine-config-daemon > /usr/local/bin/machine-config-daemon"
+  ExecStart=/usr/bin/chmod +x /usr/local/bin/machine-config-daemon
+  ExecStart=/usr/sbin/restorecon /usr/local/bin/machine-config-daemon
+
+  [Install]
+  WantedBy=multi-user.target
```

Comment:
It might be simplest to do:
```
RequiredBy=machine-config-daemon-firstboot.service
```

Then we don't need to copy all of the conditionals here - we'll only run if that unit is going to run, and also crucially it means if this unit fails then `firstboot` won't run.

Reply:
See https://github.com/openshift/installer/pull/3683 for a similar "dependency enforcement" change.